components/playlist/browser/resources/components/playlistFolder.tsx

```diff
@@ -114,11 +115,38 @@ function EditActionsContainer ({
   )
 }
 
+function useItemIdFromHash () {
+  const getId = () => window.location.hash.replace('#', '')
+  let [idFromHash, setIdFromHash] = React.useState<string>('')
+  const updateId = () => setIdFromHash(getId())
+
+  React.useEffect(() => {
+    updateId()
+    window.addEventListener('hashchange', updateId)
+    return () => window.removeEventListener('hashchange', updateId)
+  }, [])
+  return idFromHash
+}
+
+function useScrollToItem (itemId: string | undefined) {
+  const ref = React.useRef<HTMLAnchorElement>(null)
+  const scrollToElement = () => {
```

Comment:
Similar here - it'll work fine as is, but it's safer to move this directly into the effect - things get very confusing when the function/variables used in the effect aren't the same ones defined above :smile:

Effects have a lot of footguns :laughing: